backend/dataall/modules/dataset_sharing/services/share_managers/s3_access_point_share_manager.py

```diff
@@ -23,6 +23,7 @@
 IAM_ACCESS_POINT_ROLE_POLICY = "targetDatasetAccessControlPolicy"
 DATAALL_ALLOW_OWNER_SID = "AllowAllToAdmin"
 DATAALL_ACCESS_POINT_KMS_DECRYPT_SID = "DataAll-Access-Point-KMS-Decrypt"
+DATAALL_ACCESS_POINT_ENABLE_PIVOT_ROLE_PERMISSIONS_SID = "DataAll-Access-Point-Enable-Pivot-Role-Permissions"
```

Comment:
One thing I am realizing is that if a dataset has a folder share and a bucket share then there will be 2 statements enabling PivotRole Permissions in the Key Policy 

Would be curious to see if there's a way to not have duplicate here but also understand having them separate is far easier to manage in terms of checking if exists and removing if no more folder and/or bucket shares

... something to think about but deploying now to test as is and confirm the above suspicion

Reply:
I had the same thought. 
Here is how I think about it: Eventually we will move to having refactored code where we just have 1 file with all these functions consolidated in it. Currently there is a lot of common code between access point and bucket share that can be pulled out into a separate class. At that time, we can have just one of these pivot role auto create permissions in place.

Reply:
Something like a `share_managers/kms_policy_utils` similar to the `share_manager_utils`? We could have a single function that handles `update_dataset_bucket_key_policy` for both. I like that it reduces the places where we modify the policy to a single one

Reply:
@dlpzx I 100% agree we should consolidate the functions into a single utils file. However, I want to do that as a part of clean up outside of this PR. If thats okay with you, I can keep update_dataset_bucket_key_policy in both access point and bucket share for now and we can extract common functions out as a part of different issue I will create.

Reply:
In the new revision I have used KMSPivotRolePermissions as Noah advised in both the share managers.